File: messaging/sms/send-sms/code-samples/createSMSMessage.cs
// https://developers.ringcentral.com/my-account.html#/applications
// Find your credentials at the above url, set them as environment variables, or enter them below

// PATH PARAMETERS
string accountId = "<ENTER VALUE>";
string extensionId = "<ENTER VALUE>";

// POST BODY
CreateSMSMessage createSMSMessage = new CreateSMSMessage {
    from = new MessageStoreCallerInfoRequest {
        phoneNumber = "<ENTER VALUE>"
    },
    to = new[] {
        new MessageStoreCallerInfoRequest {
            phoneNumber = "<ENTER VALUE>"
        },
    },
    text = "<ENTER VALUE>"
};

RestClient rc = new RestClient(
    Environment.GetEnvironmentVariable("clientId"),
    Environment.GetEnvironmentVariable("clientSecret"),
    false
);
await rc.Authorize(
    Environment.GetEnvironmentVariable("username"),
    Environment.GetEnvironmentVariable("extension"),
    Environment.GetEnvironmentVariable("password")
);
var r = await rc.Restapi().Account(accountId).Extension(extensionId).Sms().Post(createSMSMessage);
// PROCESS RESPONSE

File: messaging/sms/create-sms-mms-message/code-samples/createSMSMessage.js
// https://developers.ringcentral.com/my-account.html#/applications
// Find your credentials at the above url, set them as environment variables, or enter them below

// PATH PARAMETERS
const accountId = '<ENTER VALUE>';
const extensionId = '<ENTER VALUE>';

// POST BODY
const body = {
    from: {
        phoneNumber: '<ENTER VALUE>'
    },
    to: [
        {
            phoneNumber: '<ENTER VALUE>'
        },
    ],
    text: '<ENTER VALUE>'
};

const SDK = require('ringcentral');
const rcsdk = new SDK({server: process.env.serverURL, appKey: process.env.clientId, appSecret: process.env.clientSecret});
const platform = rcsdk.platform();
await platform.login({ username: process.env.username, extension: process.env.extension, password: process.env.password });
const r = await platform.post(`/restapi/v1.0/account/${accountId}/extension/${extensionId}/sms`, body);
